Vizio SmartCast support

vkorn opened the pull request:
## Description:

Support of newer Vizio TV models (with SmartCast capabilities).

One annoying thing: Vizio serving https without certificate, so lot of warnings will be in the log if loglevel is low. As an option -- proxy requests through nginx. 

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#2906

## Example entry for `configuration.yaml` (if applicable):
```yaml
media_player: 
  - platform: vizio
    host: 10.0.0.1
    access_token: auth_token
```

## Checklist:

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [x] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [x] New files were added to `.coveragerc`.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

vkorn committed:
Vizio SmartCast support

vkorn committed:
Vizio SmartCast support

homeassistant commented:
Hi @vkorn, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#8260).

Once you do that we will be able to review and accept this pull request.

Thanks!

fabaff commented on homeassistant/components/media_player/vizio.py:
```diff
@@ -0,0 +1,171 @@
+    host = config.get(CONF_HOST)
+    if host is None:
+        _LOGGER.error('No host info')
+        return False
```
That's covered by the configuration validation. `CONF_HOST` will never been `None`.

fabaff commented on homeassistant/components/media_player/vizio.py:
```diff
@@ -0,0 +1,171 @@
+    token = config.get(CONF_ACCESS_TOKEN)
+    if token is None:
+        _LOGGER.error('No token info')
+        return False
```
Dito

fabaff commented on homeassistant/components/media_player/vizio.py:
```diff
@@ -0,0 +1,171 @@
+from homeassistant.helpers import config_validation as cv
+
+REQUIREMENTS = ['git+https://github.com/vkorn/pyvizio.git'
+                '@master#pyvizio==0.0.1']
```
Can you publish your module at PyPI?

fabaff commented on homeassistant/components/media_player/vizio.py:
```diff
@@ -0,0 +1,171 @@
+        return False
+    name = config.get(CONF_NAME)
+
+    add_devices([VizioDevice(host, token, name)], True)
```
Checking that the host is reachable and the token valid before setting up the platform would help that the users don't end up with a non-functional platform in their setup.

fabaff reviewed (changes requested):
Just some quick comments

vkorn committed:
Requested changes
Added new config params

vkorn committed:
Requested changes
Added new config params

armills commented:
It looks like this branch got messed up. Can you clean up the unrelated commits?

vkorn commented:
@armills , my bad, re-pushed with force. 
@fabaff , please check latest commit, I've removed double checks and introduced 2 new params. If you think that suppressing warnings is not a good idea, I'll remove it. But it's very annoying and spam to log a lot.
